rust/lance-core/src/datatypes.rs

```diff
@@ -195,20 +208,37 @@ impl TryFrom<&LogicalType> for DataType {
             match splits[0] {
                 "fixed_size_list" => {
                     if splits.len() != 3 {
-                        Err(Error::Schema {
+                        return Err(Error::Schema {
                             message: format!("Unsupported logical type: {}", lt),
                             location: location!(),
-                        })
-                    } else {
-                        let elem_type = (&LogicalType(splits[1].to_string())).try_into()?;
-                        let size: i32 = splits[2].parse::<i32>().map_err(|e: _| Error::Schema {
-                            message: e.to_string(),
-                            location: location!(),
-                        })?;
-                        Ok(FixedSizeList(
-                            Arc::new(ArrowField::new("item", elem_type, true)),
-                            size,
-                        ))
+                        });
+                    }
+
+                    let size: i32 = splits[2].parse::<i32>().map_err(|e: _| Error::Schema {
```

Comment:
In my previous attempt, i stored the fsl<bfloat16> as `fixed_size_list:fixed_size_binary:2:128`. 

Curious what would the tradeoff to keep fixed size list as 3 fields.

Reply:
Where does the extension type of the child go?

Reply:
You mean the `ARROW:extension:name` ?

https://github.com/lancedb/lance/blob/main/protos/format.proto#L382 , can we use this?

Reply:
That field is deprecated, and I think adding another meaning onto it is probably a bad idea.

The tricky part here is the fixed_size_list itself might already have an extension type. For example, in Arrow if we want `fixed_shape_tensor<bfloat16>` then we have two levels of extension types: the fixed size list has the extension type `fixed_shape_tensor`, and it's child has `lance.bfloat16`. So the `ARROW:extension:name` in a fixed size list might already be taken up; we can't record the extension type of the child there. We would need a separate field.

Reply:
Fair. 

Sounds like making sub-field of FSL and  not materialize the FSL-level offset is the long term solution then?
In that way, FSL level type is just `fixed_size_list:length`, but we don't need to store this offset array, and just store the underneath array instead?

Reply:
> Sounds like making sub-field of FSL and not materialize the FSL-level offset is the long term solution then?

Yes, though FSL-level offsets don't exist. It would just be a field without any buffers, just like how struct fields are handled today. They have children but not buffers themselves.

I started to implement that in #1693, but the page table is too fragile right now to handle this change gracefully. (See list of issues with the page table in #1809.)